internal/proxy/publish_handler_test.go

```diff
@@ -1 +1,206 @@
+	publishHandler := testDepsCfg.publishProxyHandler.Handle(node)
+
+	client, closeFn, err := centrifuge.NewClient(context.Background(), node, testDepsCfg.transport)
+	require.NoError(t, err)
+	defer func() { _ = closeFn() }()
+
+	publishReply, err := publishHandler(client, centrifuge.PublishEvent{}, rule.ChannelOptions{})
+
+	require.ErrorIs(t, centrifuge.ErrorInternal, err)
+	require.Equal(t, centrifuge.PublishReply{}, publishReply)
+}
+
+func TestHandlePublishWithProxyServerCustomDisconnect(t *testing.T) {
+	node := tools.NodeWithMemoryEngineNoHandlers()
+	defer func() { _ = node.Shutdown(context.Background()) }()
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/publish", func(w http.ResponseWriter, req *http.Request) {
+		_, _ = w.Write([]byte(`{"disconnect": {"code": 4000, "reconnect": false, "reason": "custom disconnect"}}`))
+	})
+	server := httptest.NewServer(mux)
+	defer server.Close()
+
+	testDepsCfg := newPublishHandlerTestDepsConfig(server.URL + "/publish")
+	publishHandler := testDepsCfg.publishProxyHandler.Handle(node)
+
+	client, closeFn, err := centrifuge.NewClient(context.Background(), node, testDepsCfg.transport)
+	require.NoError(t, err)
+	defer func() { _ = closeFn() }()
+
+	expectedErr := centrifuge.Disconnect{
+		Code:      4000,
+		Reason:    "custom disconnect",
+		Reconnect: false,
+	}
+	publishReply, err := publishHandler(client, centrifuge.PublishEvent{}, rule.ChannelOptions{})
+
+	require.Equal(t, expectedErr.Error(), err.Error())
+	require.Equal(t, centrifuge.PublishReply{}, publishReply)
+}
+
+func TestHandlePublishWithProxyServerCustomError(t *testing.T) {
+	node := tools.NodeWithMemoryEngineNoHandlers()
+	defer func() { _ = node.Shutdown(context.Background()) }()
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/publish", func(w http.ResponseWriter, req *http.Request) {
+		_, _ = w.Write([]byte(`{"error": {"code": 1000, "message": "custom error"}}`))
+	})
+	server := httptest.NewServer(mux)
+	defer server.Close()
+
+	testDepsCfg := newPublishHandlerTestDepsConfig(server.URL + "/publish")
+	publishHandler := testDepsCfg.publishProxyHandler.Handle(node)
+
+	client, closeFn, err := centrifuge.NewClient(context.Background(), node, testDepsCfg.transport)
+	require.NoError(t, err)
+	defer func() { _ = closeFn() }()
+
+	publishReply, err := publishHandler(client, centrifuge.PublishEvent{}, rule.ChannelOptions{})
+	expectedErr := centrifuge.Error{
+		Code:    1000,
+		Message: "custom error",
+	}
+
+	require.Equal(t, expectedErr.Error(), err.Error())
+	require.Equal(t, centrifuge.PublishReply{}, publishReply)
+}
+
+func TestHandlePublishWithInvalidCustomData(t *testing.T) {
+	node := tools.NodeWithMemoryEngineNoHandlers()
+	defer func() { _ = node.Shutdown(context.Background()) }()
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/publish", func(w http.ResponseWriter, req *http.Request) {
+		_, _ = w.Write([]byte(`{"result": {"user": "56", "expire_at": 1565436268, "b64data": "invalid data"}}`))
```

Comment:
Also result data from connect proxy, better to avoid using it here just to not confuse test readers

Reply:
fixed